rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java

```diff
@@ -216,6 +219,9 @@ private CachingRlsLbClient(Builder builder) {
       rlsChannelBuilder.disableServiceConfigLookUp();
     }
     rlsChannel = rlsChannelBuilder.build();
+    lastRlsServerConnectivityState = rlsChannel.getState(false);
+    rlsChannel.notifyWhenStateChanged(
+        lastRlsServerConnectivityState, () -> rlsServerConnectionStateChanged());
```

Comment:
This is assuming the current state, as if the state were READY, the `wasReady` would still be false because the callback wouldn't run. I think the safer thing to do is explicitly assume the current state, and the callback will run if it is different. So then you can initialize the variables assuming that initial state. `IDLE` would be a good guess for the initial state.